src/java/org/apache/cassandra/schema/TableMetadataRefCache.java

```diff
@@ -50,9 +52,9 @@ public TableMetadataRefCache(Map<TableId, TableMetadataRef> metadataRefs,
                                  Map<Pair<String, String>, TableMetadataRef> metadataRefsByName,
```

Comment:
Let's make the constructor `private` so there's no way we can create a `TableMetadataRefCache` with unprotected maps.

Reply:
@jacek-lewandowski I'd prefer to make this change as well, but didn't want to change APIs because this class exists for CASSANDRA-17044 (schema modularity). Is it safe to make this constructor private? If not, I'll make a comment expressing the caller's responsibility here.

Reply:
We could use `UnmodifiableMap.decorate` from commons collections and leave the constructor as it is. `decorate` is smarter and it would not wrap already wrapped collections.